Fixes #21: calloc int overflow causing segfault

varun-iyer opened the pull request:
This overflow occurs when the image size and the poly degree are
relatively large. As science images are frequently pretty big, it might
be worthwhile to incorporate this into the mainline, or perhaps some
documentation/a message about it.

martinberoiz commented:
I would like to merge 64d9b79 and 9a3e207, but the pull request comes with CUDA implementation, and that seems like a whole another beast to deal with. If I can figure out a way of merging only the overflow problem that'd be fine. Otherwise maybe create another PR with just those two commits? I'm new to merging PR's.

varun-iyer commented:
Thanks for pointing this out! I’m pretty new to fork/PR as well --- apparently, pushes to my master fork branch at varun-iyer/ois are automatically included in the PR. I only intended to send a PR for the first two, but it roped in my work in progress as well. In any case, I reverted the problematic commit and squashed it down, so this PR should only include the code and commits for the original fixes. The second commit will have a different hash because it technically includes the squashed reversions.

martinberoiz commented:
No problem, it's all good now and merged. Thanks for the PR!
